public/components/overview/components/select_dashboard_flyout.tsx

```diff
@@ -61,6 +63,7 @@ export function SelectDashboardFlyout({ closeFlyout, dashboardsSavedObjects, rel
   const onClickAdd = async () => {
     const selectedOption = options.find((option) => option.checked === 'on');
     if (selectedOption && selectedOption.key) {
+      getOverviewPage().createSection(DASHBOARD_SECTION);
```

Comment:
Does replace dashboard work as expected when the section is already created?

Reply:
Yes, haven't seen any problems in all my testing. Just added as a fail safe for if a user removed the datasource to ensure when it is added back it works without needing a hard refresh.